### PR TITLE
Update to Datadog client 1.*

### PR DESCRIPTION
**Pull request by barbushin**

### CODE REVIEW

**Comment by jamesgraham** [comment]
Any chance to get this approved+merged @chaseconey

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 .idea
 vendor
+composer.lock
```
Comment (chaseconey) [inline .gitignore]: Let's remove this

**Review by chaseconey — CHANGES_REQUESTED** [review]
A couple of things:

- Let's re-add the `composer.lock`, should just need to re-run `composer update` and that will get the newest version of deps
- Please fix the formatting checks that are failing as per the status checks if you could (few simple things)

Once we knock those things out, I can spin up the old test project and we can release a new major version! Thank you for the contribution, this has been on my list for awhile 👍.

**Comment by chaseconey** [comment]
Looks like we may need to downgrade our Travis setup to good ole' Trusty as well if we want to keep supporting PHP 5.4 ([src](https://travis-ci.community/t/php-5-4-and-5-5-archives-missing/3723/4)). I can do that separately, but that is why the Travis job is failing for 5.4.

**Comment by barbushin** [comment]
@chaseconey No need to add composer.lock. It's library that depends from another library that could be updated also and the main thing is to keep backward compatibility.

**Comment by barbushin** [comment]
> Please fix the formatting checks that are failing as per the status checks if you could (few simple things)

Done

**Comment by barbushin** [comment]
All fixed

**Comment by chaseconey** [comment]
@barbushin - I would like to keep the composer.lock in place in case, for whatever reason, a breaking change is introduced without a Major bump for underlying library. Been burned too many times by this.

**Comment by jamesgraham** [comment]
> @barbushin - I would like to keep the composer.lock in place in case, for whatever reason, a breaking change is introduced without a Major bump for underlying library. Been burned too many times by this.

I agree with this sentiment

**Pull request merged**